lib/components/src/blocks/ArgsTable/types.ts

```diff
@@ -0,0 +1,48 @@
+export interface JsDocParam {
+  name: string;
+  description?: string;
+}
+
+export interface JsDocReturns {
+  description?: string;
+}
+
+export interface JsDocTags {
+  params?: JsDocParam[];
+  returns?: JsDocReturns;
+}
+
+export interface PropSummaryValue {
+  summary: string;
+  detail?: string;
+  required?: boolean;
+}
+
+export type PropType = PropSummaryValue;
+export type PropDefaultValue = PropSummaryValue;
+
+export interface ArgType {
+  name?: string;
+  description?: string;
+  defaultValue?: any;
+  [key: string]: any;
+}
+
+export interface TableAnnotation {
+  type: PropType;
+  jsDocTags?: JsDocTags;
```

Comment:
I understand that you add the jsDocTags to the TableAnnotation to support legacy code but I am wondering if in the new ArgTable, there should be a JsDocAnnotation?

What's your plan for JsDoc support? I see it as a way to provide additional metadata that could be consume by the ArgsTable

Reply:
The `table` annotation is consistent with that, and I think long-term `jsDocTags` belong inside `table`. ArgTypes are very basic, like React propTypes. Annotations are used for UI-specific things. The `table` annotation contains extra info for the Props UI. The `controls` annotation is used for controls (aka knobs) specific configuration. And as more use cases come up, we can add them as needed. WDYT?

Reply:
Ok make sense, sorry I misunderstood annotation.